src/telemetry/class-parsely-telemetry.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * Parsely_Telemetry class
+ *
+ * @package Parsely
+ * @since 3.0.0
+ */
+
+declare(strict_types=1);
+
+/**
+ * This class comprises the mechanics of setting up the back end tracking instance(s).
+ * Currently, the only supported back end is Automattic Tracks.
+ * This is intended to wrap the internals such that adding / changing back ends has minimal impact on the event hooks we're interested in.
+ */
+class Parsely_Telemetry {
+	/**
+	 * This is determined by our value passed to the `WP_Widget` constructor.
+	 *
+	 * @see https://github.com/Parsely/wp-parsely/blob/e9f1b8cd1a94743e068681a8106176d23857992d/src/class-parsely-recommended-widget.php#L28
+	 */
+	const RECOMMENDED_WIDGET_BASE_ID = 'parsely_recommended_widget';
+
+	/**
+	 * Holds an instance of the class comprising the Automattic Tracks mechanics.
+	 *
+	 * @var Parsely_A8c_Tracks
+	 */
+	private $tracks;
+
+	/**
+	 * Parsely_Telemetry constructor.
+	 */
+	public function __construct() {
+		$this->tracks = new Parsely_A8c_Tracks();
```

Comment:
This highly couples the telemetry system to using the Tracks implementation - that means other systems (including mocks for tests) can't be used.

Use constructor dependency injection, type hinted at an interface, here instead.

Roughly (but you can think of better names):

```php
interface Telemetry_System {
	public function setup();
	public function record_event();
}
```

```php
class A8c_Tracks implements Telemetry_System {
	...
}
```

```php
class Telemetry {
	private $telemetry_system;

	public function __construct( Telemetry_System $telemetry_system ) {
		$this->telemetry_system = $telemetry_system;
	}
	...
}
```

```php
// At the point of instantiating Telemetry(), pass in an instance of the desired telemetry system (which at the moment, is only A8c_Tracks):
if ( apply_filters( 'wp_parsely_enable_telemetry_backend', false ) ) {
	$telemetry_system = apply_filters( 'wp_parsely_telemetry_system', new A8c_Tracks() );
	$this->telemetry = new \Parsely\Telemetry\Telemetry( $telemetry_system );
}
```

Now, folks have the ability to use a different system that extend our interface, and we can insert mocks to test the `Telemetry` class without assuming that the implementation is the 
A8c_Tracks.